app/Http/Controllers/UserController.php

```diff
@@ -67,4 +69,16 @@ public function updateUser(updateUserRequest $request, User $user)
     {
         $this->userService->updateUser($request->all(), $user);
     }
+
+    public function getUserEmails()
+    {
+        $emails = $this->userService->getUserEmails();
+        return response()->json(['result' => true, 'emails' => $emails]);
+    }
+
+    public function getUserData(GetUserDataRequest $request, User $user)
+    {
+        $user_data = $this->userService->getUserData($user);
+        return response()->json(['result' => true, 'user' => $user_data]);
```

Comment:
remove row 81. It's the redundant line. We can do something like this:
`return response()->json(new UserResource($user));`